### ISSUE TITLE
one_dashboard causes init require hashicorp/terraform

### ISSUE THREAD

**Issue by cdobbyn** [newrelic/terraform-provider-newrelic#1702]
### Terraform Version
```
Terraform v1.1.7
on linux_amd64
+ provider registry.terraform.io/newrelic/newrelic v2.40.0
```

### Affected Resource(s)
Please list the resources as a list, for example:
- `newrelic_one_dashboard`

### Terraform Configuration

```hcl
provider "newrelic" {
  account_id = "redacted"
  api_key    = var.api_key
  region     = "US"
}

resource "newrelic_one_dashboard" "default" {
  name = "default"
  page {
    name = "Page 1"

    widget_table {
      column = 0
      row    = 0
      width  = 12
      title  = "Warning Events"
      nrql_query {
        account_id = var.account_id
        query =  "SELECT * FROM InfrastructureEvent WHERE event.type = 'Warning'"
      }
    }
  }
}
```

### Actual Behavior
When running init it tries to download hashicorp/newrelic
```
Initializing provider plugins...
- Finding latest version of hashicorp/newrelic...
- Finding newrelic/newrelic versions matching ">= 2.22.0, >= 2.39.0, 2.40.0"...
- Using previously-installed newrelic/newrelic v2.40.0
╷
│ Error: Failed to query available provider packages
│ 
│ Could not retrieve the list of available versions for provider hashicorp/newrelic: provider registry registry.terraform.io does not have a provider named
│ registry.terraform.io/hashicorp/newrelic
│ 
│ Did you intend to use newrelic/newrelic? If so, you must specify that source address in each module which requires that provider. To see which modules are currently
│ depending on hashicorp/newrelic, run the following command:
│     terraform providers
╵
```

### Expected Behavior
```
Initializing provider plugins...
- Finding newrelic/newrelic versions matching ">= 2.22.0, >= 2.39.0, 2.40.0"...
- Using previously-installed newrelic/newrelic v2.40.0
```

### Steps to Reproduce
I have isolated it down to one configuration item, in order to reproduce this issue:
1. Add `account_id` to the `nrql_query` as shown above
2. `terraform init`

In order to make it go away:
1. Remove `account_id` from the `nrql_query`
3. `terraform init`

### Debug Output
At TRACE log level there's nothing useful in there.

### Panic Output
No panic.

### Important Factoids
It's directly tied to that `account_id` on the `nrql_query` and whether or not it's included.

### References
None that I can see.

**Comment by cdobbyn** [comment]
I'm not really sure what I did but I resolved this. I updated a commit on the module that was using newrelic/newrelic to not use account_id, and then I added it back in another account_id and it works properly. Super weird.

**Comment by kidk** [comment]
You may have hit a caching issue on the Terraform repository, or a delay in the release it seems. Happy it was resolved, let us know if it ever comes back 👍

**Comment by cdobbyn** [comment]
@kidk I ran into this again. I really have no idea what causes `hashicorp/newrelic` to end up in the dependency graph this but again it was tied to dashboard. In this case I moved the dashboard creation into it's own module. My repo is configured to use newrelic/newrelic. I ended up fixing the problem this time by copying the following into a versions.tf file in my module (where I hadn't previously specified the provider).

```hcl
terraform {
  required_providers {
    newrelic = {
      source  = "newrelic/newrelic"
      version = ">= 2.39.0"
    }
  }
}
```
The new version of my module just worked again. Maybe this'll help someone in the future.
